### PR TITLE
Refactor tcp_keepalive code

### PR DESCRIPTION
**Pull request by CyberDem0n**
* Move it into a separate function
* set keepalive on the REST API socket

The function will be also used in #1162

### CODE REVIEW

File: patroni/utils.py [hunk 1]
```diff
@@ -453,3 +455,38 @@ def data_directory_is_empty(data_dir):
...
+        yield (socket.SOL_TCP, 18, int(timeout * 1000))  # TCP_USER_TIMEOUT
+        TCP_KEEPIDLE = socket.TCP_KEEPIDLE
+        TCP_KEEPINTVL = socket.TCP_KEEPINTVL
+        TCP_KEEPCNT = socket.TCP_KEEPCNT
+    elif sys.platform.startswith('darwin'):
+        TCP_KEEPIDLE = 0x10  # (named "TCP_KEEPALIVE" in C)
+        TCP_KEEPINTVL = 0x101
+        TCP_KEEPCNT = 0x102
+    else:
+        return
```
Comment (erthalion) [inline patroni/utils.py:477]: Probably makes sense to add a logging message here for other systems.

**Comment by erthalion** [comment]
:

**Comment by erthalion** [comment]
:+1:

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**